packages/evm/contracts/BoostCore.sol

```diff
@@ -204,6 +227,46 @@ contract BoostCore is Ownable, ReentrancyGuard {
         return _boosts.length;
     }
 
+    /// @notice Get the incentives for a Boost
+    /// @param key The key composed of the Boost ID and the Incentive ID - keccak256(abi.encodePacked(boostId, incentiveId))
+    function getIncentive(bytes32 key) external view returns (IncentiveDisbursalInfo memory) {
+        return incentives[key];
+    }
+
+    /// @notice Returns the protocol fee and any remaining incentive value to the owner or budget
+    /// @param boostId The ID of the Boost
+    function clawback(bytes calldata data_, uint256 boostId, uint256 incentiveId) external nonReentrant {
+        // We want to check that the sender is authorized to clawback the incentive
+
+        // Generate the unique key for the incentive
+        bytes32 key = _generateKey(boostId, incentiveId);
+        IncentiveDisbursalInfo storage incentive = incentives[key];
+
+        // Decode the data for clawback
+        AIncentive.ClawbackPayload memory claim_ = abi.decode(data_, (AIncentive.ClawbackPayload));
+        uint256 amount = abi.decode(claim_.data, (uint256));
+
+        // Calculate the protocol fee based on the clawback amount and the protocol fee percentage
+        uint256 protocolFeeAmount = (amount * incentive.protocolFee) / FEE_DENOMINATOR;
+
+        // Transfer the protocol fee to the protocol fee receiver
+        if (protocolFeeAmount > 0) {
+            if (incentive.assetType == ABudget.AssetType.ERC20 || incentive.assetType == ABudget.AssetType.ETH) {
+                incentive.asset.safeTransfer(protocolFeeReceiver, protocolFeeAmount);
+            } else if (incentive.assetType == ABudget.AssetType.ERC1155) {
+                // wake-disable-next-line reentrancy (false positive, function is nonReentrant)
+                IERC1155(incentive.asset).safeTransferFrom(
+                    address(this), protocolFeeReceiver, incentive.tokenId, protocolFeeAmount, ""
```

Comment:
seems like `protocolFeeReceiver` gets the fee regardless here even on clawback. Not sure if this is intended.

Reply:
Bug - should be `claim_.target`